aten/src/ATen/native/sparse/cuda/SparseBlas.cpp

```diff
@@ -57,6 +57,13 @@ Tensor& sparse_sampled_addmm_out_sparse_csr_cuda(
 
   // there's a segfault when calling cuSPARSE on 0-sized matrices
   if (mat1.numel() == 0 || mat2.numel() == 0) {
+    // According to docs, when beta==0 values in self should be ignored.
+    // nans and infs should not propagate
+    if(beta.toComplexDouble() == 0.){
+        result.zero_();
+    } else {
+        result.mul_(beta);
+    }
```

Comment:
Optimization Nit: On second thought, why wouldn't we just move this logic in the `if` section right above where instead of `result.copy_(self)`, we do `result.copy_(beta * self)` or just `result.zero_() if beta == 0`. `self` is a vector tensor, while `result` is a matrix, so we safe on some time consuming multiplications...

Reply:
We can even go nuts and just return a broadcasted version of `beta * self` if `alpha == 0` :)